patterns/ecs-v1/grok-patterns

```diff
@@ -80,16 +80,16 @@ DATESTAMP_EVENTLOG %{YEAR}%{MONTHNUM2}%{MONTHDAY}%{HOUR}%{MINUTE}%{SECOND}
 # Syslog Dates: Month Day HH:MM:SS
 SYSLOGTIMESTAMP %{MONTH} +%{MONTHDAY} %{TIME}
 PROG [\x21-\x5a\x5c\x5e-\x7e]+
-SYSLOGPROG %{PROG:program}(?:\[%{POSINT:pid}\])?
+SYSLOGPROG %{PROG:process.program}(?:\[%{POSINT:process.pid}\])?
 SYSLOGHOST %{IPORHOST}
-SYSLOGFACILITY <%{NONNEGINT:facility}.%{NONNEGINT:priority}>
+SYSLOGFACILITY <%{NONNEGINT:log.syslog.facility.code}.%{NONNEGINT:log.syslog.priority}>
 HTTPDATE %{MONTHDAY}/%{MONTH}/%{YEAR}:%{TIME} %{INT}
 
 # Shortcuts
 QS %{QUOTEDSTRING}
 
 # Log formats
-SYSLOGBASE %{SYSLOGTIMESTAMP:timestamp} (?:%{SYSLOGFACILITY} )?%{SYSLOGHOST:logsource} %{SYSLOGPROG}:
+SYSLOGBASE %{SYSLOGTIMESTAMP:timestamp} (?:%{SYSLOGFACILITY} )?%{SYSLOGHOST:event.provider} %{SYSLOGPROG}:
```

Comment:
IIRC there is a Syslog input that instantiates a Grok filter to do its heavy-lifting, and I believe it does some magic to re-map. IMO, I'm wary of a user-instantiated filter stomping on reserved fields unless explicitly told to do so.

Reply:
continuing on the `timestamp` topic a bit, from a broader perspective.
most top level patterns capture `timestamp` - in case of syslog we could just use `timestamp` (for both timestamp formats) and let the syslog input handle the coercion and set `@timestamp`.

but there's another way we could approach timestamp - to always properly namespace (and let the user or a related plugin take it from there) : 
- e.g. in case of syslog we would always match timestamps into `system.syslog.timestamp`
  (later coerce -> rename to `@timestamp` - this is how beats approaches this)
- a matched redis log would end up with `event[redis][timestamp]`

wdyt? any preference to either name-spaced timestamps by convention or just use `timestamp` for all?

Reply:
started using `timestamp` as a convention for a field we potentially want to end up as `@timestamp`

Reply:
I think converging on this convention of always using `timestamp` would then work great with having one way to handle this `timestamp` => `@timestamp` conversion.

The date parser could default to reading from `timestamp` and output to `@timestamp`, and the docs could build on that convention to explain that convention.

I like it